python_modules/dagster/dagster_tests/scheduler_tests/test_cron_string_iterator.py

```diff
@@ -479,23 +479,85 @@ def test_cron_iterator_leap_day():
             create_pendulum_time(2023, 3, 26, 4, 0, 0, tz="Europe/Berlin"),
         ],
     ),
+    (
+        "Europe/Berlin",
+        "*/15 * * * *",
+        [
+            create_pendulum_time(2023, 3, 26, 1, 0, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 1, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 1, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 1, 45, 0, tz="Europe/Berlin"),
+            # 2 AM does not exist
+            create_pendulum_time(2023, 3, 26, 3, 0, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 3, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 3, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 3, 26, 3, 45, 0, tz="Europe/Berlin"),
+        ],
+    ),
+    (
+        "Europe/Berlin",
+        "*/15 * * * *",
+        [
+            create_pendulum_time(2023, 10, 29, 1, 0, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 1, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 1, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 1, 45, 0, tz="Europe/Berlin"),
+            create_pendulum_time(
+                2023, 10, 29, 2, 0, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 15, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 30, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 45, 0, tz="Europe/Berlin", dst_rule=pendulum.PRE_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 0, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 15, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 30, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(
+                2023, 10, 29, 2, 45, 0, tz="Europe/Berlin", dst_rule=pendulum.POST_TRANSITION
+            ),
+            create_pendulum_time(2023, 10, 29, 3, 0, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 3, 15, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 3, 30, 0, tz="Europe/Berlin"),
+            create_pendulum_time(2023, 10, 29, 3, 45, 0, tz="Europe/Berlin"),
+        ],
+    ),
 ]
 
 
 @pytest.mark.parametrize("execution_timezone,cron_string,times", DST_PARAMS)
-def test_dst_transition_advances(execution_timezone, cron_string, times):
+@pytest.mark.parametrize(
+    "force_croniter",
+    [False, True],
```

Comment:
While I think it was a good tool for developing this, are we sure we want to keep this `force_croniter` bit in? Does the 15m or maybe adding another entry or two not give us sufficient coverage? 

Forcing the code path that won't actually normally run to get coverage is subtle, I think it would take someone in the future quite some time to re-derive the reasoning. 

So I guess if we want to keep it there should be some comments explaining the thought process.